### PR TITLE
fix(capacitor): make switch statements exhaustive

### PR DESCRIPTION
**Pull request by danielsogl**
fixes: #3134

### CODE REVIEW

File: ios/Capacitor/Capacitor/Plugins/Permissions.swift [hunk 1]
```diff
@@ -61,6 +61,8 @@ public class CAPPermissionsPlugin: CAPPlugin {
         ret = "denied"
       case .authorized:
         ret = "granted"
+    case .limited:
+        ret = "granted"
```
Comment (danielsogl) [inline ios/Capacitor/Capacitor/Plugins/Permissions.swift:65]: @jcesarmobile Maybe we should return `limited` as result?

File: ios/Capacitor/Capacitor/Plugins/Permissions.swift [hunk 2]
```diff
@@ -97,6 +99,8 @@ public class CAPPermissionsPlugin: CAPPlugin {
         ret = "denied"
       case .notDetermined:
         ret = "prompt"
+      case .ephemeral:
+        ret = "granted"
```
Comment (danielsogl) [inline ios/Capacitor/Capacitor/Plugins/Permissions.swift:103]: @jcesarmobile Maybe we should return `ephemeral` as result?

**Comment by dwieeb** [comment]
@danielsogl It looks like the iOS build is failing for this PR, would you be able to look into it?

**Comment by danielsogl** [comment]
@dwieeb  I think the error is exclusive to iOS 14 because the new given options are not available for iOS 13.

**Comment by ikeith** [comment]
Yes, the additional cases will break compilation under previous versions of Xcode. This needs some` #available` checks to conditionally reference the new cases only when building with the beta(s).

**Comment by danielsogl** [comment]
I will update the PR

File: ios/Capacitor/Capacitor/Plugins/Permissions.swift [hunk 3]
```diff
@@ -97,8 +101,11 @@ public class CAPPermissionsPlugin: CAPPlugin {
         ret = "denied"
       case .notDetermined:
         ret = "prompt"
+      #if swift(>=5.3)
+      case .ephemeral:
+        ret = "ephemeral"
       }
-      
+      #endif
```
Comment (ikeith) [inline ios/Capacitor/Capacitor/Plugins/Permissions.swift:108]: This `#endif` needs to be moved up. When compiled out, it's breaking the brackets on the switch statement.

**Comment by ikeith** [comment]
After some examination, the problem is a bit more complicated. I've opened a new PR, #3160, to clean things up with the enums. Please feel free to comment on that if you have any concerns.

**Pull request closed without merging**